eclipse/che
# Added checksums to configmaps to enable pod restart on configmap change.

### What does this PR do?
Currently when you modify the configmap (like adding a new env variable), or change the value of an env variable (change the value in a values.yaml), the pod will not be restarted and you have to manually do a rollout restart.  This can introduce a configuration mismatch and it is not in line with k8s/helm philosophy.  A change in the config should automatically issue a restart.

This PR adds a checksum value in an annotation so that if the configmap changes, the spec of the pod will also change and k8s will trigger a restart.  This is a [documented way (see Automatically Roll Deployments section)](https://helm.sh/docs/howto/charts_tips_and_tricks/) of fixing that issue.

### How to test this PR?
Using the helm installer, simply change a value that is used in the configmap and the che pod (or the plugin/devfile registries) should restart.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
